var/spack/repos/builtin/packages/elpa/package.py

```diff
@@ -76,7 +73,17 @@ def setup_environment(self, spack_env, run_env):
         spack_env.set('SCALAPACK_LDFLAGS', spec['scalapack'].libs.joined())
```

Comment:
I'm just getting familiar with Spack, and trying to debug elpa on my systems (#6143, #6144). I'm wondering if the use of `setup_environment()` is appropriate in this package. According to #2016: the purpose of `setup_environment()` seems to be setting up environment modules, not setting up environment for `configure()` or `build()`.

Reply:
See above, that's what the reasoning of @alalazo who move it from configure to `setup_environment()`

> To group together all the changes to the environment variables in the function that is meant to change the environment variables for this package.

Reply:
> According to #2016: the purpose of setup_environment() seems to be setting up environment modules

it's called twice, https://github.com/spack/spack/issues/2016#issuecomment-258936647 so it's not only for modules

Reply:
Thanks for the clarification. If those environment variables are not needed in modulefiles, can they be moved to `configure_args()`? Could you please take a look at my tentative changes [here](https://github.com/baip/spack/commit/af4ed0b0c700c6444fbab022cd1adf501c04fb19)? With these changes, my issues with GCC 6.4.0/7.2.0 (#6143) or Intel 2018.0 (#6144) went away. Before, in the second call to `setup_environment()`, `spec['mpi'].mpicc` somehow became undefined.

Reply:
Please open a PR with suggested changes so that everyone can comment